hwdetect/hwdetect.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+function echoDetected() {
+    
+    if [ -s $2 ]
+    then
+        echo $1:
+        cat $2
+    else
+        echo No $1 detected.
+    fi
+}
+
+function echoDiff {
+    
+    ls /dev | grep $2 > temp_diff.txt
+    local tmp_diff=$(diff temp_diff.txt $1)
+    
+    if [ ! -z "$tmp_diff" ]
+    then
+        echo $tmp_diff
+        ls /dev | grep $2 > $1
```

Comment:
The operation is done twice during one diff cycle, and it can return different results.
Copy `temp_diff.txt` from the above line to `$1`

Reply:
Fixed